mongo/client.go

```diff
@@ -613,7 +613,15 @@ func (c *Client) configure(opts *options.ClientOptions) error {
 		if err := opts.ServerAPIOptions.ServerAPIVersion.Validate(); err != nil {
 			return err
 		}
-		c.serverAPI = *opts.ServerAPIOptions
+
+		// manually clone the passed in options to avoid future modifications of original ServerAPIOptions.
```

Comment:
Nit: This reads as if this copying prevents the original ServerAPIOptions from being modified. I suggest saying something like: "... so future modifications of the original ServerAPIOptions have no effect".

Reply:
Changed to `manually clone the passed in options so future modifications of the original ServerAPIOptions have no effect.`